src/test/java/teammates/storage/api/NotificationsDbTest.java

```diff
@@ -0,0 +1,272 @@
+package teammates.storage.api;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.test.BaseTestCaseWithLocalDatabaseAccess;
+
+/**
+ * SUT: {@link NotificationsDb}.
+ */
+public class NotificationsDbTest extends BaseTestCaseWithLocalDatabaseAccess {
+
+    private final NotificationsDb notificationsDb = NotificationsDb.inst();
+    private final Map<String, NotificationAttributes> typicalNotifications = getTypicalDataBundle().notifications;
+
+    // Notification attributes will be assigned with value at setup()
+    private NotificationAttributes n;
+
+    @BeforeMethod
+    public void setup() throws Exception {
+        for (NotificationAttributes n : typicalNotifications.values()) {
+            notificationsDb.createEntity(n);
+        }
+        n = typicalNotifications.get("notification1");
+    }
+
+    /**
+     * Removes all notifications created by each test.
+     */
+    @AfterMethod
+    public void cleanUp() {
+        List<NotificationAttributes> retrieved = notificationsDb.getAllNotifications();
+        assertNotNull(retrieved);
+
+        retrieved.forEach(n -> notificationsDb.deleteNotification(n.getNotificationId()));
+    }
+
+    @Test
+    public void testGetNotification() throws Exception {
+        ______TS("typical success case");
+        NotificationAttributes retrieved = notificationsDb.getNotification(n.getNotificationId());
+        assertNotNull(retrieved);
```

Comment:
```suggestion
        NotificationAttributes actual = notificationsDb.getNotification(n.getNotificationId());
        assertEquals(expected.toString(), actual.toString());
```

Need to check actual == expected, the expected is from the typical data bundle, the actual is from DB

For variable names, try to use `expected`, `actual` when applicable to make it clearer.

Reply:
Noted. Thank you for the comment.

Reply:
One is list and the other one is set so it is not that easy to do direct `==`. I will use contains to compare every element between the two instead.